ios/RNModelRepository.mm

```diff
@@ -0,0 +1,39 @@
+
+#import "RNModelRepository.h"
+#import "RNPixelBufferUtilities.h"
+
+#import "TIOModelRepository.h"
+#import "TIOModelBundle.h"
+#import "TIOModelUpdater.h"
+
+@interface RNModelRepository()
+
+@property TIOModelRepository* repository;
+
+@end
+
+@implementation RNModelRepository
+
+RCT_EXPORT_MODULE();
+
+RCT_EXPORT_METHOD(initialize:(NSString*)baseUrl) {
+    NSURL *URL = [NSURL URLWithString:baseUrl];
+    self.repository = [[TIOModelRepository alloc] initWithBaseURL:URL session:nil];
+}
+
+RCT_EXPORT_METHOD(updateModel:(NSString*)pathToFile callback:(RCTResponseSenderBlock)callback) {
+    //Get model bundle from url
+    TIOModelBundle *bundle = [[TIOModelBundle alloc] initWithPath:pathToFile];
+
+    //Use updator to actually update model
+    TIOModelUpdater *updater = [[TIOModelUpdater alloc] initWithModelBundle:bundle repository:self.repository];
```

Comment:
The risk here is that, because an update involves asynchronous code, the `updater` instance may fall out of scope and be deallocated before it executes that callback, preventing the update's callback from ever being called.

We'll probably have to set the updater as an instance variable like you're doing with `repository`, and then nil it out inside the callback block.

Problem is you'll then also have to check if an updater already exists before creating another one, and if one does, error out, so that you don't overwrite the instance variable, which would deallocate the previous updater instance and prevent it from executing the callback block. 

Which means you can only update one model at a time.

Reply:
Good idea. 👍